packages/jest-watcher/src/types.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {Config, TestResult} from '@jest/types';
+
+type TestSuiteInfo = {
+  config: Config.ProjectConfig;
+  duration?: number;
+  testPath: string;
+};
+
+export type JestHookExposedFS = {
+  projects: Array<{
+    config: Config.ProjectConfig;
+    testPaths: Array<Config.Path>;
+  }>;
+};
+
+export type FileChange = (fs: JestHookExposedFS) => void;
+export type ShouldRunTestSuite = (
+  testSuiteInfo: TestSuiteInfo,
+) => Promise<boolean>;
+export type TestRunComplete = (results: TestResult.AggregatedResult) => void;
+
+export type JestHookSubscriber = {
+  onFileChange: (fn: FileChange) => void;
+  onTestRunComplete: (fn: TestRunComplete) => void;
+  shouldRunTestSuite: (fn: ShouldRunTestSuite) => void;
+};
+
+export type JestHookEmitter = {
+  onFileChange: (fs: JestHookExposedFS) => void;
+  onTestRunComplete: (results: TestResult.AggregatedResult) => void;
+  shouldRunTestSuite: (testSuiteInfo: TestSuiteInfo) => Promise<boolean>;
+};
+
+export type UsageData = {
+  key: string;
+  prompt: string;
+};
+
+export type AllowedConfigOptions = Partial<
```

Comment:
basically https://github.com/facebook/jest/blob/dc355000db72a6bbf52ce7d28620497873926c94/packages/jest-cli/src/lib/update_global_config.js#L14-L36

Reply:
Huh, as I look a this, they're actually wrong. I took my list from the docs, but this is longer than that.

Reply:
https://github.com/facebook/jest/pull/7843/commits/9950466e2672314f70ae4d339f10a6f9a509098b